WebARKit/WebARKitTrackers/WebARKitOpticalTracking/WebARKitTracker.cpp

```diff
@@ -444,66 +463,88 @@ class WebARKitTracker::WebARKitTrackerImpl {
             assert(initialized == false);
         }
 
-        WEBARKIT_LOGi("Reset Tracking!\n");
+        if (_trackVizActive) {
+            memset(_trackViz.bounds, 0, 8 * sizeof(float));
+            _trackViz.opticalFlowTrackablePoints.clear();
+            _trackViz.opticalFlowTrackedPoints.clear();
+            _trackViz.opticalFlowOK = false;
+        }
+
+        WEBARKIT_LOGd("Reset Tracking!\n");
 
         clear_output();
 
         _isDetected = false;
 
         cv::Mat frameDescr;
         std::vector<cv::KeyPoint> frameKeyPts;
-        bool valid;
 
-        cv::Mat featureMask = createFeatureMask(frame);
+        // This if cond. doesn't works as expected in artoolkitx. This make the tracking process unstable.
+        // if (_currentlyTrackedMarkers < _maxNumberOfMarkersToTrack) {
```

Comment:
strangely if i add this line it will not track properly, mybe i missed some code... i will look into it later.